src/components/container/SwellingCalendar/SwellingCalendar.tsx

```diff
@@ -0,0 +1,177 @@
+import { useEffect, useState } from "react";
+import MyDataHelps, { Guid, SurveyAnswer, SurveyAnswersQuery } from "@careevolution/mydatahelps-js";
+import React from "react";
+import surveyResults from "../../../helpers/get-survey-answers";
+import Calendar from "../../presentational/Calendar/Calendar";
+import { parseISO, startOfDay, startOfMonth } from "date-fns";
+import { Button, CalendarDay, CalendarDayStateConfiguration, DateRangeNavigator, LoadingIndicator } from "../../presentational";
+import { previewSwellingData } from "./SwellingCalendar.previewdata";
+import language from "../../../helpers/language";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeSvgIcon } from "react-fontawesome-svg-icon";
+export type SwellingCalendarPreviewState = "WithData" | "NoData" | "Loading" | "Live";
+
+export interface SwellingCalendarProps {
+    surveyName : string,
+    dateRecordedResultIdentifier : string, 
+    severityResultIdentifier : string, 
+    previewState? : SwellingCalendarPreviewState,
+}
+
+interface SwellingDay {
+    dateRecorded : string,
+    severity : string
+}
+
+export default function (props: SwellingCalendarProps) {
+    const surveyAnswerQuery : SurveyAnswersQuery = {
+        surveyName : props.surveyName,
+        resultIdentifier : [props.dateRecordedResultIdentifier, props.severityResultIdentifier]
+    }
+    const [datePagerMonth, setMonth] = useState<Date>(startOfMonth(new Date()));
+    const [data, setData] = useState<Map<string, SwellingDay> | undefined>();
+
+    async function initialize(){
+        if (!data){
+            if (props.previewState && props.previewState == "NoData")
+            {
+                transformToCalendarData([]);
+            } else {
+                if (props.previewState && props.previewState == "WithData"){
+                    transformToCalendarData(previewSwellingData);
+                }
+                else
+                {
+                    surveyResults(surveyAnswerQuery).then((results: SurveyAnswer[]) => {
+                        transformToCalendarData(results);
+                    });
+                }
+            }
+        }
+    }
+
+    const stateConfiguration: CalendarDayStateConfiguration = {
+        'mild': {
+            style: {
+                background: '#F7DA19'
+            },
+            streak: true,
+            streakColor: '#FDEF97'
+        },
+        'moderate': {
+            style: {
+                background: 'var(--mdhui-color-warning)'
+            },
+            streak: true,
+            streakColor: '#FCCCC7'
+        },
+        'severe': {
+            style: {
+                background: 'var(--mdhui-color-danger)'
+            },
+            streak: true,
+            streakColor: '#FCCCC7'
+        },
+    };
+
+    const computeStateForDay = (date: Date): string => {
+        let key : string = startOfDay(date).toDateString();
+        return (data && data?.get(key)?.severity) ?? "";
```

Comment:
Minor:  You don't need the `data &&` part here, which will also let you get rid of the extra set or parenthesis.